scrape/scrape.go

```diff
@@ -544,6 +601,34 @@ func (sp *scrapePool) refreshTargetLimitErr() error {
 	return err
 }
 
+func verifyLabelLimits(lset labels.Labels, limits labelLimits) error {
+	met := lset.Get(labels.MetricName)
+
+	if limits.labelLimit != 0 {
+		nbLabels := len(lset)
+		if nbLabels > int(limits.labelLimit) {
+			return fmt.Errorf("label_limit exceeded (metric: %.50s, number of label: %d, limit: %d)", met, nbLabels, limits.labelLimit)
+		}
+	}
+
+	for _, l := range lset {
+		if limits.labelNameLengthLimit != 0 {
```

Comment:
we could check that once before iterating over the labels

Reply:
I added a check to verify that both limits are set before entering the loop, but we will still need to keep the checks inside of the loop for the comparisons.